aiinfra-cluster/installation_scripts/install_cloud_ops_agent.sh

```diff
@@ -85,7 +217,11 @@ main() {
         fail "Legacy or Ops Agent is already installed."
     fi
 
-    install_opsagent
+    echo "Install Ops Agent"
+    retry_with_backoff 10 32 install_opsagent
+
+    echo "Install DCGM"
+    install_dcgm
```

Comment:
since we are doing `wget` while installing DCGM, should we retry with backoff as well ?

Reply:
I think it would be beneficial to have the retry logic for both the `wget` and the `apt-get` parts of the install since I have seen the `apt-get` portion of the install_opsagent be a point of failure; however, it seems incorrect(/wasteful?) to have the other stuff like the `systemctl start/restart` commands within the retry block. I reworked the function a bit if you could take a look at it and see if I am thinking about this wrong or just being too pedantic.